src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Parser class to parse the input commands by the user.
+ */
+public class Parser {
+    String command;
+    Ui ui;
+    Storage storage;
+    TaskList tasks;
+
+    enum Activity {
+        TODO, DONE, EVENT, DELETE, DEADLINE, BYE, LIST, NORMAL, FIND
+    }
+
+    /**
+     * Public constructor for Parser.
+     *
+     * @param command The input command of the user.
+     * @param ui The Ui to return messages to the user.
+     * @param storage To load or save tasks by the user.
+     * @param tasks The list of tasks input by the user.
+     */
+    public Parser(String command, Ui ui, Storage storage, TaskList tasks) {
+        this.command = command;
+        this.ui = ui;
+        this.storage = storage;
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns a boolean, whether a program should end or not.
+     *
+     * @return True if program should still run, false otherwise.
+     */
+    public boolean getRun() {
+        return !storage.isExit();
+    }
+
+    /**
+     * Parses the user command and inputs the task accordingly.
+     *
+     * @throws DukeException If task input by user is incomplete.
+     * @throws DeleteException If delete is incomplete.
+     * @throws IOException If an input or output operation is failed or interpreted.
+     */
+    public void parseCommand() throws DukeException, DeleteException, IOException, FindException {
+
+        Activity activity;
+        if (command.equals("bye")) {
```

Comment:
This method / while loop might be too long. This [coding quality website](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)  stated the following:

>Be wary when a method is longer than the computer screen, and take corrective action when it goes
>beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

This is a little longer than 30 LOC , though I too am not sure how to condense this part of logic haha, maybe implementing a 'command' class?

Reply:
I agree with @Ongzl. Perhaps, you can consider moving the content in the switch cases to separate methods to shorten this method body?

Reply:
hmm, may i clarify something? 30 LOC is for one method right? Technically I have many different switch cases, by which each case has << 30 LOC (the longest switch case has 15 LOC inclusive of empty lines). So a method refers to one switch case right, not the entire switch?